From 16bd709ff20252291d656fde39c17ec188f259c0 Mon Sep 17 00:00:00 2001
From: affggh <879632264@qq.com>
Date: Tue, 15 Jul 2025 14:48:33 +0800
Subject: [PATCH 1/2] build: fix compile warnings

---
 tools/kallsym.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/kallsym.c b/tools/kallsym.c
index 9506002..3b89a30 100644
--- a/tools/kallsym.c
+++ b/tools/kallsym.c
@@ -713,7 +713,7 @@ static int correct_addresses_or_offsets_by_vectors(kallsym_t *info, char *img, i
     } else {
         info->kallsyms_addresses_offset = pos;
         tools_logi("kallsyms_addresses offset: 0x%08x\n", pos);
-        tools_logi("kernel base address: 0x%08llx\n", info->kernel_base);
+        tools_logi("kernel base address: 0x%08llx\n", (unsigned long long)info->kernel_base);
     }
 
     return 0;
@@ -778,7 +778,7 @@ static int correct_addresses_or_offsets_by_banner(kallsym_t *info, char *img, in
         info->kallsyms_addresses_offset = pos;
         tools_logi("kallsyms_addresses offset: 0x%08x\n", pos);
         info->kernel_base = uint_unpack(img + info->kallsyms_addresses_offset, elem_size, info->is_be);
-        tools_logi("kernel base address: 0x%llx\n", info->kernel_base);
+        tools_logi("kernel base address: 0x%llx\n", (unsigned long long)info->kernel_base);
     }
 
     int32_t pid_vnr_offset = get_symbol_offset(info, img, "pid_vnr");
-- 
2.43.0

